apollo-router/src/plugins/authorization/authenticated.rs

```diff
@@ -353,20 +326,19 @@ impl<'a> transform::Visitor for AuthenticatedVisitor<'a> {
 
     fn fragment_spread(
         &mut self,
-        node: &hir::FragmentSpread,
-    ) -> Result<Option<apollo_encoder::FragmentSpread>, BoxError> {
-        let fragments = self.compiler.db.fragments(self.file_id);
-        let condition = fragments
-            .get(node.name())
-            .ok_or("MissingFragmentDefinition")?
-            .type_condition();
+        node: &ast::FragmentSpread,
+    ) -> Result<Option<ast::FragmentSpread>, BoxError> {
+        let condition = &self
+            .fragments
+            .get(&node.fragment_name)
+            .ok_or("MissingFragment")?
+            .type_condition;
```

Comment:
unrelated to this PR again, but what would happen if all the selections under a fragment were removed by the authz filter? Shouldn't we remove the spread then?

Reply:
Yes we probably should :sweat_smile: